### PR TITLE
Training of Multilingual Keyphrase Extraction model

### PR DESCRIPTION
**Pull request by AmitChaulwar**
This PR includes a DistilBert2Joint network for training a multilingual keyphrase extraction model using `distilbert-base-multilingual-cased` as a base model. The multilingual dataset can be used from https://github.com/LIAAD/KeywordExtractor-Datasets. The preprocessing scripts are also added.

### CODE REVIEW

File: scripts/model.py [hunk 1]
```diff
@@ -6,109 +6,135 @@
...
-        loss = self.network(**inputs)
-
-        if self.args.n_gpu > 1:
-            # mean() to average on multi-gpu parallel (not distributed) training
-            loss = loss.mean()
-            
-        if self.args.gradient_accumulation_steps > 1:
-            loss = loss / self.args.gradient_accumulation_steps
-            
+        with torch.cuda.amp.autocast():
```
Comment (lumalik) [inline scripts/model.py:80]: cool, did not know this exists!

File: preprocess/prepro_utils.py [hunk 2]
```diff
@@ -3,6 +3,14 @@
 import logging
 import unicodedata
 from nltk.stem.porter import PorterStemmer
+import string
+import spacy
+from nltk.stem import SnowballStemmer
+
+# Load spacy model for stemming
+stemmer_spanish = SnowballStemmer("spanish")
```
Comment (lumalik) [inline preprocess/prepro_utils.py:11]: why load the spanish news dataset here and the Spanish stemmer?

File: preprocess/spllit_json.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+"""Split the json data in train, test and validation data"""
```
Comment (lumalik) [inline preprocess/spllit_json.py:1]: if the name of the file a typo?

**Review by lumalik — APPROVED** [review]
LGTM! Left some minor comments

**Comment by SunSiShining** [comment]
Amit adds multilingual adaptation and fp16 training mode. They are very useful for more users.

Thanks, Amit!

File: preprocess/prepro_utils.py (hunk 2 above)
Comment (SunSiShining) [reply preprocess/prepro_utils.py:11]: Yes, I think it's an upload error.

This becomes a bug because line 293 (prepro_utils.py) shows that even the English dataset uses the Spanish stemmer.

I have solved this problem. Thanks Lukas!
